### PR TITLE
JAMES-2586 Fix PostgresEmailGetMethodTest, PostgresEmailQueryMethodTest, PostgresMailboxSetMethodTest

### PR DESCRIPTION
**Pull request by hungphan227**

### CODE REVIEW

**Comment by Arsnael** [comment]
Timeout in the fork. That's our culprit. 

Also wasn't this commit https://github.com/apache/james-project/pull/2048/commits/1b16d50f350532d3459cb05a8e669f684ec779fc part of this work? Or not?

**Comment by hungphan227** [comment]
> Timeout in the fork. That's our culprit.
> 
> Also wasn't this commit [1b16d50](https://github.com/apache/james-project/commit/1b16d50f350532d3459cb05a8e669f684ec779fc) part of this work? Or not?

Ah yeah. But I temporarily removed it to check other commits

File: mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresAttachmentModule.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public interface PostgresAttachmentModule {
     interface PostgresAttachmentTable {
 
         Table<Record> TABLE_NAME = DSL.table("attachment");
-        Field<UUID> ID = DSL.field("id", SQLDataType.UUID.notNull());
+        Field<String> ID = DSL.field("id", SQLDataType.VARCHAR.notNull());
```
Comment (chibenwa) [inline mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresAttachmentModule.java:38]: :-(

Comment (hungphan227) [reply mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresAttachmentModule.java:38]: https://github.com/apache/james-project/pull/2048#discussion_r1502017082
https://github.com/apache/james-project/pull/2048#discussion_r1502032559

Comment (chibenwa) [reply mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresAttachmentModule.java:38]: How about being clever and reworking `AttachmentId`?

We could have:
 - An interface with a Factory for `AttachmentId`
 - Current AttachmentId class be renamed as` LegacyAttachmentId` and used by Cassandra app, JPA, memory
 - Have a new `UuidBackedAttachmentId` used in the Postgres app ?

**Comment by chibenwa** [comment]
Rationnals: the string field is NOT space efficient. 

attachment table is a big one, and a criteria for success of the PG app is to succeed to keep the DB small.

File: mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresAttachmentModule.java (hunk 1 above)
Comment (hungphan227) [reply mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresAttachmentModule.java:38]: Move this issue to another ticket https://github.com/orgs/linagora/projects/2/views/2?pane=issue&itemId=56534403
This PR stay for too long and should be closed soon

**Review by quantranhong1999 — APPROVED** [review]

File: server/protocols/jmap-rfc-8621-integration-tests/postgres-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/postgres/PostgresEmailQueryMethodTest.java [hunk 2]
```diff
@@ -19,10 +19,65 @@
...
+    @Disabled("Flaky test. TODO stabilize it.")
+    public void listMailsShouldBeSortedWhenUsingFrom(GuiceJamesServer server) {
+    }
 
-@Disabled
-// TODO Need to fix
-public class PostgresEmailQueryMethodTest extends PostgresBase implements EmailQueryMethodContract {
+    @Override
+    @Test
+    @Disabled("Flaky test. TODO stabilize it.")
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/postgres-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/postgres/PostgresEmailQueryMethodTest.java:80]: For thos failing tests in Email query method, it seems more code related in general, as we can see similar issues I think from this PR on distributed tests: https://github.com/apache/james-project/pull/2116

Comment (Arsnael) [reply server/protocols/jmap-rfc-8621-integration-tests/postgres-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/postgres/PostgresEmailQueryMethodTest.java:80]: I would be ok to solve it in a separate ticket, with master fixes included if needed too.

Comment (hungphan227) [reply server/protocols/jmap-rfc-8621-integration-tests/postgres-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/postgres/PostgresEmailQueryMethodTest.java:80]: I have created ticket for few weeks https://github.com/linagora/james-project/issues/5092

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Merging this for now, as tickets have been created to tackle the rest.

**Pull request merged**